src/UnitySampleProject/Assets/GelatoTest.cs

```diff
@@ -0,0 +1,240 @@
+using System.Numerics;
+using System.Threading;
+using System.Threading.Tasks;
+using ChainSafe.GamingSDK.EVM.MetaMaskBrowserWallet;
+using ChainSafe.GamingWeb3.Build;
+using ChainSafe.GamingWeb3.Unity;
+using ChainSafe.GamingSdk.Gelato;
+using ChainSafe.GamingSdk.Gelato.Dto;
+using ChainSafe.GamingSdk.Gelato.Types;
+using ChainSafe.GamingWeb3;
+using Microsoft.Extensions.DependencyInjection;
+using Nethereum.Hex.HexTypes;
+using UnityEngine;
+using Web3Unity.Scripts.Library.Ethers.JsonRpc;
+using Web3Unity.Scripts.Library.Ethers.Transactions;
+using Contract = Web3Unity.Scripts.Library.Ethers.Contracts.Contract;
+
+public class GelatoTest : MonoBehaviour
+{
+    private Web3 _web3;
+
+    // Start is called before the first frame update
+    async void Start()
+    {
+        await InitWeb3();
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+    }
+
+    public async void TestingGelatoFunction()
+    {
+        Debug.Log("Testing works");
+    }
+
+    private async Task InitWeb3()
+    {
+        var sponsorApiKey = System.Environment.GetEnvironmentVariable("1BalanceApiKey");
+
+        _web3 = await new Web3Builder(ProjectConfigUtilities.Load()).Configure(services =>
+        {
+            services.UseUnityEnvironment();
+            services.UseJsonRpcProvider();
+            services.UseWebPageWallet();
+            services.UseGelatoModule(sponsorApiKey);
+        }).BuildAsync();
+    }
+
+    public async void CallWithSyncFeeExample()
+    {
+        var vitalik = "0xd8da6bf26964af9d7eed9e03e53415d37aa96045";
+        var target = "0xA045eb75e78f4988d42c3cd201365bDD5D76D406";
+        var feeToken = "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE";
+        var abi = "[{\"inputs\": [" +
+                  "{\"internalType\":\"address\",\"name\":\"_token\",\"type\":\"address\"}," +
+                  "{\"internalType\":\"address\",\"name\":\"_to\",\"type\":\"address\"}," +
+                  "{\"internalType\":\"uint256\",\"name\":\"_amount\",\"type\":\"uint256\"}" +
+                  "]," +
+                  "\"name\":\"sendToFriend\"," +
+                  "\"outputs\":[]," +
+                  "\"stateMutability\":\"nonpayable\"," +
+                  "\"type\":\"function\"" +
+                  "}]";
+        var contract = new Contract(abi, target);
+
+        var data = contract.Calldata("sendToFriend", new object[]
+        {
+            feeToken,
+            vitalik,
+            new BigInteger(5 * 10E12)
+        });
+        var gelatoInstance = _web3.ServiceProvider.GetRequiredService<IGelatoModule>();
```

Comment:
Should be replaced with `web3.GelatoTranscationExecutor.SendTransaction(new GelatoTransactionRequest(...))`, as we discussed.
In a different branch perhaps?

Reply:
Sorry I'm not sure what the executor you're referring to is or where it should exist

Reply:
We have ITransactionExecutor with a single method `Task<TransactionResponse> SendTransaction(TransactionRequest transaction)`. With the introduction of the Gelato module, we want to have IGelatoTransactionExecutor with a similar method, but with a different parameter type GelatoTransactionRequest. This type would contain all the additional information required for Gelato transaction execution. Not sure about the return type, it might be the same.
IGelatoTransactionExecutor would go to the core library and also be added as a property to the Web3 class.
GelatoTransactionExecutor would go to the Gelato module library.
It should also be bound when you call UseGelato.